scripts/rename-cjs.mjs

```diff
@@ -0,0 +1,15 @@
+/** Renames `.js` files in the /lib/cjs folder to `.cjs` to appease ESM clients. */
+
+import { promises as fs } from "node:fs";
+import url from "node:url";
+
+process.chdir(url.fileURLToPath(`${import.meta.url}/../../dist/_cjs`));
+for (const filename of await fs.readdir(".")) {
+	if (!/\.js$/.test(filename)) continue;
+	const cjs = await fs.readFile(filename, "utf-8");
+	await fs.writeFile(
+		filename.replace(/\.js$/, ".cjs"),
+		cjs.replaceAll(/require\("(.\/[\w\-.]+)\.js"\)/g, 'require("$1.cjs")'),
+	);
+	await fs.unlink(filename);
+}
```

Comment:
I tried comparing your previous version + without namespace + using tsup (https://antfu.me/posts/publish-esm-and-cjs#tsup)

https://github.com/oasisprotocol/deoxysii-js/commit/32e7271652ac6e2194b22d186c7d413e7da16c26

might be useful some day

Reply:
tsup looks really good, and I really don't like the `rename-cjs` hack. I wish there was a simpler way that only needed a single set of files.

Unfortunately there seems to be no safe way to drop CommonJS support

Reply:
```
$ pnpm tsup src/index.ts  --format cjs,esm --dts --clean --minify --sourcemap
CLI Building entry: src/index.ts
CLI Using tsconfig: tsconfig.json
CLI tsup v8.0.2
CLI Target: es2021
CLI Cleaning output folder
CJS Build start
ESM Build start
ESM dist/index.mjs     10.17 KB
ESM dist/index.mjs.map 68.06 KB
ESM ⚡️ Build success in 21ms
CJS dist/index.js     10.69 KB
CJS dist/index.js.map 68.32 KB
CJS ⚡️ Build success in 22ms
DTS Build start
DTS ⚡️ Build success in 723ms
DTS dist/index.d.ts  1.17 KB
DTS dist/index.d.mts 1.17 KB
```

`tsup` is great, so much less hassle.